src/rinex.c

```diff
@@ -880,7 +880,40 @@ static int decode_obsdata(FILE *fp, char *buff, double ver, int mask,
             }
         }
     }
-    
+    if (m>=2) {
+        if (val[l[0]]==0.0&&val[l[1]]==0.0) {
+            p[l[0]]=-1; p[l[1]]=-1;
+        }
+        else if (val[l[0]]!=0.0&&val[l[1]]==0.0) {
+            p[l[0]]=1; p[l[1]]=-1;
+        }
+        else if (val[l[0]]==0.0&&val[l[1]]!=0.0) {
+            p[l[0]]=-1; p[l[1]]=1;
+        }
+        else if (ind->pri[l[1]]>ind->pri[l[0]]) {
+            p[l[1]]=1; p[l[0]]=NEXOBS<2?-1:NFREQ+1;
+        }
+        else {
+            p[l[0]]=1; p[l[1]]=NEXOBS<2?-1:NFREQ+1;
+        }
+    }
+    if (q>=2) {
+        if (val[r[0]]==0.0&&val[r[1]]==0.0) {
+            p[r[0]]=-1; p[r[1]]=-1;
+        }
+        else if (val[r[0]]!=0.0&&val[r[1]]==0.0) {
+            p[r[0]]=2; p[r[1]]=-1;
+        }
+        else if (val[r[0]]==0.0&&val[r[1]]!=0.0) {
+            p[r[0]]=-1; p[r[1]]=2;
+        }
+        else if (ind->pri[r[1]]>ind->pri[r[0]]) {
+            p[r[1]]=2; p[r[0]]=NEXOBS<3?-1:NFREQ+2;
+        }
+        else {
+            p[r[0]]=2; p[r[1]]=NEXOBS<3?-1:NFREQ+2;
+        }
+    }
```

Comment:
This difference needs to be account for, it might be that this was a fix in demo5 that has not made it to qt_gui and  merging this would be a regression?

This is the commit on demo5:
commit ffe990d6a1c251cc2cc744b6fc46e725928437c2
Author: rtklibexplorer <tim.everett3@gmail.com>
Date:   Tue Jan 2 16:00:49 2024 -0700

    - Switch Beidou code priorities to match recent switch of 3rd solution frequency
    - Fix bug in handling priorities with multiple codes